### PR TITLE
Improve logs

### PR DESCRIPTION
**Pull request by nholland94**
I went through all the logging statements and made sure they all used metadata interpolation instead of string interpolation. This helps when categorizing logs from a metrics and alerting standpoint. I also added some metadata to a couple of logs and attempted to standardize more of the metadata names, which helps with correlating logs in queries.

### CODE REVIEW

**Comment by nholland94** [comment]
Closes #4527

**Comment by Schmavery** [comment]
heroic

File: src/lib/child_processes/termination.ml [hunk 1]
```diff
@@ -19,9 +19,9 @@ let check_terminated_child t child_pid logger =
   if Pid.Table.mem t child_pid then (
     let kind = Pid.Table.find_exn t child_pid in
     Logger.error logger ~module_:__MODULE__ ~location:__LOC__
-      "Child process of kind $process_kind with pid $child_pid has \
-       terminated; terminating parent process $pid"
+      "Child process of kind $process_kind with pid $child_pid has terminated"
       ~metadata:
-        [ ("child_pid", `Int (Pid.to_int child_pid))
+        [ ("parent_pid", `Int (Pid.to_int (Unix.getpid ())))
```
Comment (psteckler) [inline src/lib/child_processes/termination.ml]: I believe the metadata contains `pid` by default, and that this entry duplicates that information (though maybe it's helpful to have the name `parent_pid`)

File: src/lib/coda_metrics/coda_metrics.ml [hunk 2]
```diff
@@ -759,9 +759,8 @@ let server ~port ~logger =
   let open Cohttp_async in
   let handle_error _ exn =
     Logger.error logger ~module_:__MODULE__ ~location:__LOC__
-      !"ecountered error while handling request to prometheus server: %{sexp: \
-        Error.t}"
-      (Error.of_exn exn)
+      ~metadata:[("error", `String (Exn.to_string exn))]
+      !"ecountered error while handling request to prometheus server: $error"
```
Comment (ghost-not-in-the-shell) [inline src/lib/coda_metrics/coda_metrics.ml]: Maybe we can get rid of this exclamation mark.

File: src/lib/coda_metrics/coda_metrics.ml [hunk 3]
```diff
@@ -759,9 +759,8 @@ let server ~port ~logger =
   let open Cohttp_async in
   let handle_error _ exn =
     Logger.error logger ~module_:__MODULE__ ~location:__LOC__
-      !"ecountered error while handling request to prometheus server: %{sexp: \
-        Error.t}"
-      (Error.of_exn exn)
+      ~metadata:[("error", `String (Exn.to_string exn))]
+      "ecountered error while handling request to prometheus server: $error"
```
Comment (Schmavery) [inline src/lib/coda_metrics/coda_metrics.ml]:
```suggestion
      "Encountered error while handling request to prometheus server: $error"
```
not that it's a big deal either way

**Pull request merged**